### PR TITLE
chore: bump ibc-go and ibctest to v6

### PR DESCRIPTION
**Pull request by colin-axner**
updating ibc-go and ibctest to v6, running into an issue of indirect dependency on ibc-go v5

Requires: https://github.com/strangelove-ventures/lens/pull/239

### CODE REVIEW

File: ibctest/poll_test.go [hunk 1]
```diff
@@ -42,7 +42,7 @@ func pollForUpdateClient(ctx context.Context, chain ibc.Chain, startHeight, maxH
 		return nil, errors.New("not found")
 	}
 
-	bp := test.BlockPoller{CurrentHeight: chain.Height, PollFunc: doPoll}
```
Comment (colin-axner) [inline ibctest/poll_test.go:45]: see TODO on line 17

File: ibctest/relayer.go [hunk 2]
```diff
@@ -80,8 +80,8 @@ func (r *Relayer) AddChainConfiguration(ctx context.Context, _ ibc.RelayerExecRe
 	return nil
 }
 
-func (r *Relayer) AddKey(ctx context.Context, _ ibc.RelayerExecReporter, chainID, keyName string) (ibc.Wallet, error) {
-	res := r.sys().RunC(ctx, r.log(), "keys", "add", chainID, keyName)
+func (r *Relayer) AddKey(ctx context.Context, _ ibc.RelayerExecReporter, chainID, keyName, coinType string) (ibc.Wallet, error) {
+	res := r.sys().RunC(ctx, r.log(), "keys", "add", chainID, keyName, "--coin-type", fmt.Sprint(coinType))
```
Comment (colin-axner) [inline ibctest/relayer.go:84]: I took what was in ibctest to update the coin type usage

File: ibctest/relayer.go [hunk 3]
```diff
@@ -121,6 +121,30 @@ func (r *Relayer) UpdatePath(ctx context.Context, _ ibc.RelayerExecReporter, pat
 	return nil
 }
 
+func (r *Relayer) GetClients(ctx context.Context, rep ibc.RelayerExecReporter, chainID string) (ibc.ClientOutputs, error) {
```
Comment (colin-axner) [inline ibctest/relayer.go:124]: implemented based on GetChannels

**Comment by colin-axner** [comment]
Requires: https://github.com/strangelove-ventures/lens/pull/239, using fork as proof of concept of testing passing. Will update the commit when the lens pr is merged

**Comment by colin-axner** [comment]
Please feel free to takeover this pr. I'll be away for quite awhile after Dec 15

**Comment by agouin** [comment]
Thanks for this @colin-axner. We went ahead and updated main to ibc-go v7 though to prepare for tm v0.37 event encoding differences. Closing this as stale.

**Pull request closed without merging**